app/views/posts/new.html.haml

```diff
@@ -1,7 +1,10 @@
 - content_for(:page_title) do
   %h1 Share New Knowledge
 
-= simple_form_for @post do |f|
+= simple_form_for @post, :html => {:multipart => true} do |f|
   = f.input :title
   = f.input :body
+  = f.label "Upload an image"
```

Comment:
This should be:

    = f.input :image, label: "Upload an image", hint: "For best results, choose an image that's at least 400 pixels wide and at least 150 pixels tall."

The way you have it now, the label isn't actually hooked up to the input field.